### DISCUSSION TITLE
Can't install dokan library because previous version exists.

### DISCUSSION THREAD

**Discussion by Weroxig** [dokan-dev/dokany#1257] [Q&A]
[Dokan_Library_2.2.0.1000_Bundle_20241231203657.log](https://github.com/user-attachments/files/18280540/Dokan_Library_2.2.0.1000_Bundle_20241231203657.log)
[Dokan_Library_2.2.0.1000_Bundle_20241231203701.elevated.log](https://github.com/user-attachments/files/18280541/Dokan_Library_2.2.0.1000_Bundle_20241231203701.elevated.log)
[Dokan_Library_2.2.0.1000_Bundle_20241231203657_000_Dokan_x64.msi.log](https://github.com/user-attachments/files/18280542/Dokan_Library_2.2.0.1000_Bundle_20241231203657_000_Dokan_x64.msi.log)

search string "dokan" returns no results in `C:\Windows` I've also deleted all registry keys with the word "dokan" in it.
the dokan2 driver is also removed. `sc delete dokan2` has already been run.

**Comment by Liryna** [comment]
You are correct that the previous install is still there. You might be able to remove it by running with admin permission `msiexec.exe /x {9A7325EA-D3C9-0202-0000-240818095850}` or (vers 2.2.0.1000) `msiexec.exe /x Dokan_x64.msi`
```
MSI (s) (00:B0) [20:37:04:281]: Product: Dokan Library 2.2.0.1000 (x64) -- A previous version of Dokan Library 2.2.0.1000 (x64) is already installed. Please uninstall the existing version first, reboot the system and start the setup again.

A previous version of Dokan Library 2.2.0.1000 (x64) is already installed. Please uninstall the existing version first, reboot the system and start the setup again.
Action ended 20:37:04: LaunchConditions. Return value 3.
Action ended 20:37:04: INSTALL. Return value 3.
```

**Reply by Weroxig** [reply]
`msiexec.exe /x {9A7325EA-D3C9-0202-0000-240818095850}`
and
`msiexec.exe /x Dokan_x64.msi` on the 2.2.0.1000 installer both give this error unfortunately.

![image](https://github.com/user-attachments/assets/06e89b2d-3d60-4b71-8ad1-db84814abb2b)

and doesn't let the installer install dokany still.

**Reply by Liryna** [reply]
Seems like you have an older Dokan version 2 installed which needs to be uninstalled first. Maybe ccleaner can help here or another app that allow to force uninstall apps

**Comment by R084N6** [comment]
This was an old discussion but it is still an issue so I wanted to add some information in case it may help someone else. I work tech support for a software that uses Dokan Library in the background and my users are running into this. The Dokan install fails causing our product install to fail. If there was a previous install of Dokan v2.*the newer Dokan v2.* will fail unless the previous version is cleanly uninstalled. It does not appear to care if there is a Dokan v1.* still installed. I don't know if the install could be rewritten to also ignore different v2 installs like it does with the v1, but that would be helpful. Unfortunately there does not seem to be a one size fits all fix for it.

In testing, I have sometimes been able to simply use the Windows "Add or Remove Programs" to uninstall "Dokan Library 2.*" then reboot the machine and run the newer installer. It has not always worked though. There are sometimes leftovers in the registry that will still prevent the new install. I usually use a tool such as Revo Uninstaller for all my uninstalls as it has options to delete junk Windows tends to leave in the Registry and File System. They do have a free version, and I think CCleaner can also do this, but I have not used it recently. This method also needs a reboot after uninstalling.

I have had some users report that method failing as well. For some of them, next going into the registry via Regedit (please backup your registry before doing this, just in case), searching for and deleting anything "dokan," then rebooting has allowed the install to work. Still others have reported even that has not been sufficient to clean it up enough for an install. Some have tried all of the above but still had a service stuck and were able to get rid of it by opening an Admin command prompt window and running;
sc delete dokan2
